components/rich_text_editor/rich_text_editor.vue

```diff
@@ -0,0 +1,250 @@
+     * - `Number` Sets the focus to a specific position in the input
+     * - `true`   Defaults to `start`
+     * - `false`  Disables autofocus
+     * @values true, false, start, end, all, number
+     */
+    autofocus: {
+      type: [Boolean, String, Number],
+      default: false,
+      validator (autofocus) {
+        if (typeof autofocus === 'string') {
+          return RICH_TEXT_EDITOR_AUTOFOCUS_TYPES.includes(autofocus);
+        }
+        return true;
+      },
+    },
+
+    /**
+     * The output format that the editor uses when emitting the "@input" event.
+     * One of `text`, `json`, `html`. See https://tiptap.dev/guide/output for
+     * examples.
+     * @values text, json, html
+     */
+    outputFormat: {
+      type: String,
+      default: 'text',
+      validator (outputFormat) {
+        return RICH_TEXT_EDITOR_OUTPUT_FORMATS.includes(outputFormat);
+      },
+    },
+  },
+
+  emits: [
+    /**
+     * Editor input event
+     * @event input
+     * @type {String|JSON}
+     */
+    'input',
+
+    /**
+     * Event to sync the value with the parent
+     * @event input
+     * @type {String|JSON}
+     */
+    'update:value',
+
+    /**
+     * Editor blur event
+     * @event blur
+     * @type {FocusEvent}
+     */
+    'blur',
+
+    /**
+     * Editor focus event
+     * @event focus
+     * @type {FocusEvent}
+     */
+    'focus',
+  ],
+
+  data () {
+    return {
+      editor: null,
+    };
+  },
+
+  computed: {
+    extensions () {
+      // These are the default extensions needed jsut for plain text.
+      const extensions = [Document, Paragraph, Text];
+      // TODO: Add logic for extensions passed as props.
+      return extensions;
+    },
+
+    inputAttrs () {
+      const attrs = {
+        'aria-label': this.inputAriaLabel,
+        'aria-multiline': true,
+        role: 'textbox',
+      };
+      if (!this.editable) {
+        attrs['aria-readonly'] = true;
+      }
+      return attrs;
+    },
+  },
+
+  /**
+    * Because the Editor instance is initialized when mounted it does not get
+    * updated props automatically, so the ones that can change after mount have
+    * to be hooked up to the Editor's own API.
+    */
+  watch: {
+    editable (isEditable) {
+      this.editor.setEditable(isEditable);
+      this.updateEditorAttributes({ 'aria-readonly': isEditable });
+    },
+
+    inputClass (newClass) {
+      this.updateEditorAttributes({ class: newClass });
+    },
+
+    inputAriaLabel (newLabel) {
+      this.updateEditorAttributes({ 'aria-label': newLabel });
+    },
+
+    value (newValue) {
+      if (newValue === this.getOutput()) {
+        // The new value came from this component and was passed back down
+        // through the parent, so don't do anything here.
+        return;
+      }
+      // Otherwise replace the content (resets the cursor position).
+      this.editor.commands.setContent(newValue, false);
+    },
+  },
+
+  mounted () {
+    this.createEditor();
```

Comment:
Should be possible to do this on created rather than mounted right since the editor-content component is what's binding to the DOM not the editor itself? Not that big of deal but might give a slight performance increase.

Reply:
Yeah fair enough, moved this to created() instead